### PR TITLE
Add script to find fast mirror for docker

### LINKED ISSUE #2339: Make build process faster in China
**Is your feature request related to a problem? Please describe.**
We need a way to make build process faster in China

**Describe the solution you'd like**
change ubuntu docker software update source
change maven repo
change go proxy
change docker mirror

**Describe alternatives (optional)**

**Additional Notes**

### PR DESCRIPTION
**Pull request by lhw362950217**
Fixes: #2339

### CODE REVIEW

**Review by Yancey0623 — DISMISSED** [review]
LGTM

**Comment by wangkuiyi** [comment]
@lhw362950217 The CI log warns https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/340332267#L677-L679 that we should use minikube driver docker instead of none.  The docker driver doesn't require root privilege. I tried minikube with docker driver locally on a Linux VM, it seems that we need to add the user to the Linux group named docker to remove the requirement of root privilege. We might need to add `sudo usermod -aG docker $USER`.

File: docker/dev/find_fastest_resources.sh [hunk 1]
```diff
@@ -131,3 +134,29 @@ function find_fastest_go_proxy() {
...
+function find_fastest_docker_url() {
+	read -r -d '' download_urls <<-EOM
+		https://get.daocloud.io/docker
+		https://get.docker.com
+	EOM
+	find_fastest_url $download_urls
+}
+
+# Find fastest docker mirror url
+function find_fastest_docker_mirror() {
```
Comment (wangkuiyi) [inline docker/dev/find_fastest_resources.sh:147]: docker_mirror => docker_registry ?

File: scripts/travis/start_argo.sh [hunk 2]
```diff
@@ -14,10 +14,21 @@
 
 set -e
 
+# If argo already installed, skip
+if kubectl get namespace argo>/dev/null 2>&1; then
+	exit 0
```
Comment (wangkuiyi) [inline scripts/travis/start_argo.sh:20]: Add an echo here?

```bash
echo "Argo is already installed."
```

**Review by wangkuiyi — COMMENTED** [review]
LGTM

**Comment by lhw362950217** [comment]
> @lhw362950217 The CI log warns https://travis-ci.com/github/sql-machine-learning/sqlflow/jobs/340332267#L677-L679 that we should use minikube driver docker instead of none. The docker driver doesn't require root privilege. I tried minikube with docker driver locally on a Linux VM, it seems that we need to add the user to the Linux group named docker to remove the requirement of root privilege. We might need to add `sudo usermod -aG docker $USER`.

Yes, I also encounter this warning. I will try to change the driver.

**Review by Yancey0623 — DISMISSED** [review]

**Review by Yancey0623 — APPROVED** [review]

**Pull request merged**